@msh/msh.m

```diff
@@ -1572,9 +1576,44 @@ function write(obj,fname,type)
             [p1(:,1),p1(:,2)] = m_ll2xy(p1(:,1),p1(:,2)) ;
             [p2(:,1),p2(:,2)] = m_ll2xy(p2(:,1),p2(:,2)) ;
             if nfix > 0
-                [pfixx(:,1),pfixx(:,2)] = m_ll2xy(pfixx(:,1),pfixx(:,2)) ;
+                [pfixx(:,1),pfixx(:,2)] = m_ll2xy(pfixx(:,1),pfixx(:,2));
+            end
+
+            % checking for weirs in obj1 to keep
+            if any(obj1.bd.ibtype == 24)
+                disp('Weirs found in obj1, extracting to ensure they are preserved')
+                % get the weir indices
+                weir_nodes = [];
+                for ii = 1:obj1.bd.nbou
+                    if obj1.bd.ibtype(ii) ~= 24; continue; end 
+                    nodes = full(obj1.bd.nbvv(1:obj1.bd.nvell(ii),ii));
+                    nodes2 = full(obj1.bd.ibconn(1:obj1.bd.nvell(ii),ii));
+                    weir_nodes = [weir_nodes; nodes; nodes2];
+                end
+                % retrieve elements connected to weir nodes
+                vtoe = VertToEle(t1); 
+                connec_eles = vtoe(:,weir_nodes);
+                connec_eles(connec_eles == 0) = [];
+                connec_eles = unique(connec_eles)';
+                tw = t1(connec_eles,:);
+                % retrieve all other elements
+                unconnec_eles = setdiff([1:length(t1)]',connec_eles);
+                tn = t1(unconnec_eles,:);
+                p1o = p1;
+                [p1,t1] = fixmesh(p1,tn);
+                % ensure that the new obj1 is traversable
+                objn = msh(); objn.t = t1; objn.p = p1; 
+                objn = Make_Mesh_Boundaries_Traversable(objn,0,1);
+                % add the deleted elements into the weir triangulation
+                twadd = setdiff(t1,objn.t,'rows');
+                [~,~,I] = intersect(p1(twadd,:),p1o,'rows','stable');
+                twadd = reshape(I,3,3);
```

Comment:
This line causes a segfault for me because the length of I cannot be reshaped into a 3 by 3 matrix. Is this by accident? I'm trying to understand the logic of this. It seems you remove the weir nodes/elements from the inset mesh and then do the merger. Later on you put the nodes and weirs back on?